stores/hostdb.go

```diff
@@ -919,16 +915,20 @@ func (ss *SQLStore) processConsensusChangeHostDB(cc modules.ConsensusChange) {
 
 	var newAnnouncements []announcement
 	for _, sb := range cc.AppliedBlocks {
-		// Fetch announcements and add them to the queue.
 		var b types.Block
 		convertToCore(sb, &b)
-		hostdb.ForEachAnnouncement(b, height, func(hostKey types.PublicKey, ha hostdb.Announcement) {
-			newAnnouncements = append(newAnnouncements, announcement{
-				hostKey:      publicKey(hostKey),
-				announcement: ha,
+
+		// Process announcements, but only if they are not too old.
+		if b.Timestamp.After(time.Now().UTC().Add(-ss.announcementMaxAge)) {
```

Comment:
Not 100% sure but I think if you use `After` or `Before` the library will implicitly handle the timezone and you don't need `UTC()`.